tests/test_3qbits_gates.py

```diff
@@ -74,13 +74,19 @@ def check_ccz_with_heralds_or_ancillaries_and_get_performance(processor, herald_
                 assert pytest.approx(modulus, error) == modulus_value
                 assert modulus != 0
 
+                # if i_state != data_state:
```

Comment:
In this file you can replace the call to `check_ccz_with_heralds_or_ancillaries_and_get_performance` (and deleted the method) from your new one in tests/test_controlled_gates.py `check_controlled_gates_and_get_performance`.
Maybe you can delete the file test_3qbits_gates and bring back all tests from this file into your file ?